lib/ddtrace/contrib/http/circuit_breaker.rb

```diff
@@ -5,24 +5,52 @@ module HTTP
       # HTTP integration circuit breaker behavior
       # For avoiding recursive traces.
       module CircuitBreaker
-        def should_skip_tracing?(req, address, port, transport, pin)
-          # we don't want to trace our own call to the API (they use net/http)
-          # when we know the host & port (from the URI) we use it, else (most-likely
+        def should_skip_tracing?(req, address, port, tracer)
+          return true if datadog_http_request?(req, address, port, tracer)
```

Comment:
I was thinking about this concept - and I wonder if safer, and lower overhead approach would be to make a copy of `Net::HTTP` before it has been instrumented and use that class as a transport mechanism for sending traces to the agent.

irb snippet showing the concept of creating a class copy and using its original copy without the changes:
```
2.4.1 :001 > class A; def x; puts 'x'; end; end
 => :x
2.4.1 :002 > B = A.dup
 => B
2.4.1 :003 > A.class_eval do def x; puts 'a'; end; end
 => :x
2.4.1 :004 > A.new.x
a
 => nil
2.4.1 :005 > B.new.x
x
```

Would it make sense to use this pattern here ? Or do you find some cases where that would be more complex than existing solution ?

Reply:
It sounds complex.

Ultimately the issue with this entire circuit breaker is that the `Net::HTTP` integration is assuming what the transport is, and how it submits traces; there's no way it can or should know this. (This is a problem with the design in general, not your suggestion.)

The best solution and replacement for this would be a "don't trace" block that the transport can activate itself to disable tracing for its operations. This would do a few things: 1) eliminate this circuit breaker from `Net::HTTP` entirely, 2) allow circuit breaking to be applied generally against any instrumented code the transport might run through, not just `Net:HTTP`. I would like to add this functionality in a future PR.

Reply:
>The best solution and replacement for this would be a "don't trace" block that the transport can activate itself to disable tracing for its operations.

Its definitely possible to implement and would also lead to a cleaner code that what is currently here, so seems reasonable 👍